WorkbookDataStore refactoring

mXaln opened the pull request:
Splits up & relocates the code in WorkbookDatastore into appropriate classes.

<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/Bible-Translation-Tools/Orature/742)
<!-- Reviewable:end -->

mXaln committed:
Remove localeLanguage injection

mXaln committed:
Removed audio related stuff from WorkbookDataStore

mXaln committed:
Removed appPreferences dependency from WorkbookDataStore

mXaln committed:
Fixed bug

mXaln committed:
Remove playback speed rate from WorkbookDataStore

mXaln committed:
Removed WorkbookDataStore from IDependencyGraphProvider

mXaln committed:
Removed activeResourceMetadata from WorkbookDataStore

jsarabia committed:
get new audio player as a utility method

Rather than get the audio player directly from the dependency graph everywhere, we create a utility method.

jsarabia committed:
Move initializeProjectFiles to HomePageViewModel

Rather than have this function globally accessible (when it should only be called once) it is moved to the ViewModel where the function was called.

jsarabia committed:
Remove updateSelectedTakes from workbookdatastore

AnonymousWalker commented on jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/screens/chunking/ChunkingDebugMain.kt:
```diff
@@ -65,8 +65,8 @@ class ChunkingDebugView : View(){
-        val wkbk: WorkbookDataStore = tornadofx.find()
-        wkbk.sourceAudioProperty.set(SourceAudio(File("/Users/jsara/Documents/test12345.mp3"), 0, 1))
+        val audio: AudioDataStore = tornadofx.find()
+        audio.sourceAudioProperty.set(SourceAudio(File("/Users/jsara/Documents/test12345.mp3"), 0, 1))
```
/path/to/file.mp3  ?

AnonymousWalker commented on jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/AppPreferencesStore.kt:
```diff
@@ -0,0 +1,57 @@
+
+    private fun updatePlaybackSpeedRate(event: PlaybackRateChangedEvent) {
+        when (event.type) {
+            PlaybackRateType.SOURCE -> workbookDataStore.workbook.translation.updateSourceRate(event.rate)
```
can we have a method in workbookDS to avoid this "train wreck" code on these lines?

It's also mentioned in Clean Code, FYI
https://wiki.c2.com/?TrainWreck

AnonymousWalker commented on jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/AudioDataStore.kt:
```diff
@@ -0,0 +1,176 @@
+                }
+            }
+            chapter != null -> { /* no-op */
+            } // preserve targetAudio for clean up
```
the placement of this comment seems not right to me. We should put it inside/next to the block similar to `/* no-op */`

AnonymousWalker commented on jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/AudioDataStore.kt:
```diff
@@ -0,0 +1,176 @@
+            val chunk = workbookDataStore.activeChunkProperty.get()
+            val chapter = workbookDataStore.activeChapterProperty.get()
+            if (chapter != null && chunk != null) {
+                sourceAudioProperty.set(workbook.sourceAudioAccessor.getChunk(chapter.sort, chunk.sort, workbook.target))
```
when a method call is getting long and nested like this, I usually introduce a variable so that it's easier to know what value is set

```
val ___ = workbook.sourceAudioAccessor.getChunk(chapter.sort, chunk.sort, workbook.target)
sourceAudioProperty.set(___)
```

AnonymousWalker commented on jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/ExportChapterViewModel.kt:
```diff
@@ -72,7 +72,7 @@ class ExportChapterViewModel : ViewModel() {
         Completable
             .fromAction {
-                workbookDataStore.activeProjectFilesAccessor.setContributorInfo(contributors)
+                workbookDataStore.workbook.projectFilesAccessor.setContributorInfo(contributors)
```
another "train wreck"?

AnonymousWalker reviewed (changes requested):
great work! It feels a lot cleaner now

jsarabia committed:
Update ChunkingDebugMain.kt

jsarabia committed:
Move audio data store tests out of workbook data store tests

jsarabia committed:
Remove resources and fix mocks

Removes resources (which was already set for removal) rather than fixing up related tests

Corrects mocks as the project files accessor has been moved into the workbook
